x-pack/plugins/apm/public/context/apm_service_context.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext, ReactNode } from 'react';
+import { isRumAgentName } from '../../common/agent_name';
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../common/transaction_types';
+import { useServiceTransactionTypes } from '../hooks/use_service_transaction_types';
+import { useUrlParams } from '../hooks/useUrlParams';
+import { useServiceAgentName } from '../hooks/use_service_agent_name';
+import { IUrlParams } from './UrlParamsContext/types';
+
+export const APMServiceContext = createContext<{
+  agentName?: string;
+  transactionType?: string;
+  transactionTypes: string[];
+}>({ transactionTypes: [] });
+
+export function ApmServiceContextProvider({
+  children,
+}: {
+  children: ReactNode;
+}) {
+  const { urlParams } = useUrlParams();
+  const { agentName } = useServiceAgentName();
+  const transactionTypes = useServiceTransactionTypes();
+  const transactionType = getTransactionType({
+    urlParams,
+    transactionTypes,
+    agentName,
+  });
+
+  return (
+    <APMServiceContext.Provider
+      value={{ agentName, transactionType, transactionTypes }}
+      children={children}
+    />
+  );
+}
+
+export function getTransactionType({
```

Comment:
I've always had a dilemma about exporting or not a function only to unit test it. Couldn't you run the same unit tests but using  `ApmServiceContextProvider` or maybe `useApmService` as the entry point? 

PS: I'm not saying that it's wrong, I just feel bad about it 🤪

Reply:
Good point. I'll look into it 👍